Doc/library/sqlite3.rst

```diff
@@ -49,81 +49,102 @@ This document includes four main sections:
 
 .. _sqlite3-tutorial:
 
+.. We use the following practises for SQL code:
+.. - UPPERCASE for keywords
+.. - lowercase for schema
+.. - single quotes for string literals
+.. - singular for table names
+.. - if needed, use double quotes for table and column names
+
 Tutorial
 --------
 
-To use the module, start by creating a :class:`Connection` object that
-represents the database.  Here the data will be stored in the
-:file:`example.db` file::
+In this tutorial, you will learn the basics of the :mod:`!sqlite3` API
+by creating an on-disk database :file:`python.db`
+for managing Python release metadata.
+A fundamental understanding of database concepts,
+like `transactions`_ and `cursors`_, is assumed.
+
+To start working with an SQLite database,
+you must first open a connection to it.
+Call the :func:`sqlite3.connect` function
+to create a :class:`Connection` to the database :file:`python.db`::
 
    import sqlite3
-   con = sqlite3.connect('example.db')
+   con = sqlite3.connect("python.db")
 
-The special path name ``:memory:`` can be provided to create a temporary
-database in RAM.
+SQLite will implicitly create the database file :file:`python.db`
+in the current working directory if it does not exist.
+The returned :class:`Connection` object ``con``
+represents our connection to the on-disk database.
 
-Once a :class:`Connection` has been established, create a :class:`Cursor` object
-and call its :meth:`~Cursor.execute` method to perform SQL commands::
+In order to execute SQL statements and fetch results from SQL queries,
+we need to use a database cursor.
+Call the :meth:`~Connection.cursor` method of ``con`` to create
+:class:`Cursor` ``cur``::
 
    cur = con.cursor()
 
-   # Create table
-   cur.execute('''CREATE TABLE stocks
-                  (date text, trans text, symbol text, qty real, price real)''')
-
-   # Insert a row of data
-   cur.execute("INSERT INTO stocks VALUES ('2006-01-05','BUY','RHAT',100,35.14)")
-
-   # Save (commit) the changes
-   con.commit()
+Now, let us create a database table ``release`` with columns for version
+number and release date, and inserting the release data for Python 1.0.
+Execute the SQL statements
+by calling the :meth:`~Cursor.execute` method of ``cur``::
 
-   # We can also close the connection if we are done with it.
-   # Just be sure any changes have been committed or they will be lost.
-   con.close()
+   cur.execute("""CREATE TABLE release(version, releasedate)""")
```

Comment:
SQLite lets us create columns without data types; I deliberately exploit that feature to make the SQL compact. I'm not sure it makes sense to mention it in the actual docs, though.

Reply:
We definitely don't want to go into any real detail and its best to simplify it as much as practical; on the other hand, I wouldn't want it to be _too_ magical and bite users later, if there isn't at least some link/reference to what's really going on. But I'm not the SQL expert to judge this this; I've probably learned as much about it from reviewing your PRs as anywhere else :laughing:

Reply:
I've mentioned this feature; PTAL